### DISCUSSION TITLE
Cannot reproduce visual effects

### DISCUSSION THREAD

**Discussion by LitterWindwind** [NVlabs/instant-ngp#588] [Q&A]
When I compare the JSON file provided by nerf dataset with the JSON file generated by colmap2nerf, the visual effect is very different, so is the effect of colmap reconstruction very bad?
![image](https://user-images.githubusercontent.com/35981299/167535039-b1e06dcc-7e11-4b4e-bd38-e305db8000f0.png)
![image](https://user-images.githubusercontent.com/35981299/167535067-c690df4c-7c86-48fd-988a-ed452a112953.png)

**Comment by Tom94** [comment]
Colmap does its best to reconstruct camera poses from 2D images. This is an incredibly difficult task, so there will naturally be some error, especially if it is used on low-resolution, non-real-world data such as the synthetic scenes.

In contrast, the camera poses bundled with the synthetic scenes are known to be perfect, because they were used to generate the synthetic images in the first place.

So yes, colmap2nerf is bound to be worse for the synthetic scenes. You should not use it when you can get accurate camera poses by some other means. For real-world scenes, colmap is one of the best at what it does, though.
